From oneukum@suse.de Wed Mar 14 03:11:20 2007
From: Oliver Neukum <oneukum@suse.de>
To: Greg KH <gregkh@suse.de>
Subject: USB: necessary update for mos7720 driver
Date: Wed, 14 Mar 2007 11:11:08 +0100
Message-Id: <200703141111.09064.oneukum@suse.de>

From: Oliver Neukum <oneukum@suse.de>

these devices have a shared interrupt endpoint. For serialcore to pass
an interrupt endpoint to a subdriver, the subdriver must define and
_export_ a fitting callback. The mos7720 driver failed to do so. This led
invariably to an oops upon open. This patch fixes it. The driver is useless
without it. Please try to get this into 2.6.21 and the stable kernels that
have this driver.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/mos7720.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/serial/mos7720.c
+++ b/drivers/usb/serial/mos7720.c
@@ -1628,6 +1628,7 @@ static struct usb_serial_driver moschip7
 	.chars_in_buffer	= mos7720_chars_in_buffer,
 	.break_ctl		= mos7720_break,
 	.read_bulk_callback	= mos7720_bulk_in_callback,
+	.read_int_callback	= mos7720_interrupt_callback,
 };
 
 static int __init moschip7720_init(void)
